finality-aleph/src/new_network/connection_manager/session.rs

```diff
@@ -265,39 +293,74 @@ mod tests {
 
     #[tokio::test]
     async fn creates_with_local_address() {
+        let (authority_index_and_pen, authority_verifier) =
+            keyboxes_components().await.pop().unwrap();
         assert!(Handler::new(
-            keyboxes().await.pop().unwrap(),
+            authority_index_and_pen,
+            authority_verifier,
             SessionId(43),
             local_p2p_addresses()
         )
         .await
         .is_ok());
     }
 
+    #[tokio::test]
+    async fn creates_without_node_index_nor_authority_pen() {
```

Comment:
You should also test somewhere that such a handler returns `None` from `authentication()`.

Reply:
Done

Reply:
I cannot find this, did you forget to commit it or am I blind (we already recently saw that the latter is quite probable ;p)?

Reply:
This time it is my mistake, forgot to push :sweat_smile: